runtimes/node-18.0/server.js

```diff
@@ -18,25 +18,51 @@ const server = micro(async (req, res) => {
     };
 
     const contentType = req.headers['content-type'] ?? 'text/plain';
-    const rawBody = await parseText(req);
-    let body = rawBody;
+    const bodyString = await parseText(req);
+    let body = bodyString;
 
     if (contentType.includes('application/json')) {
-        body = await parseJson(req);
+        if(bodyString) {
+            body = await parseJson(req);
+        } else {
+            body = {};
+        }
     }
 
     const headers = {};
     Object.keys(req.headers).filter((header) => !header.toLowerCase().startsWith('x-open-runtimes-')).forEach((header) => {
         headers[header.toLowerCase()] = req.headers[header];
     });
 
+    const scheme = (req.headers['x-forwarded-proto'] ?? 'http');
+    const host = req.headers['host'].includes(':') ? req.headers['host'].split(':')[0] : req.headers['host'];
+    const port = +(req.headers['host'].includes(':') ? req.headers['host'].split(':')[1] : '80');
```

Comment:
whats the + supposed to do ?

Reply:
cast from string to integer.

Reply:
okay 👍 . This can be avoided if we use the std library

Reply:
Ill use std library.